### PR TITLE
Populate pickup/dropoff type and booking field from pattern halt

### PR DESCRIPTION
**Pull request by binh-dam-ibigroup**
### Checklist

- [x] Appropriate branch selected => This changes a flex feature.
- [na] Any modified or new methods or classes have helpful JSDoc and code is thoroughly commented
- [na] The description lists all applicable issues this PR seeks to resolve
- [na] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

In the timetable editor, when adding a new stop time to a pattern (*not* duplicating an existing one),
the value in fields `pickupType`, `dropOffType`, `pickupBookingRuleId`, `dropOffBookingRuleId` are incorrectly left as null.
This PR populates these fields from the pattern halt as desired.

### CODE REVIEW

**Comment by br648** [comment]
@binh-dam-ibigroup This is failing serverside. I think this is because gtfs-lib (dev-flex) has this update: https://github.com/conveyal/gtfs-lib/pull/369 where as datatool-ui (dev-flex) does not have this update: https://github.com/ibi-group/datatools-ui/pull/890. Phil should be able to help here.

cc @philip-cline

**Comment by philip-cline** [comment]
There are quite a few merge conflicts between dev and dev-flex relating mostly to the stop_headsigns and continuous_pickup/dropoff rework. Let me know if you want me to make the merge or if you'll tackle it @binh-dam-ibigroup

**Comment by philip-cline** [comment]
I'll create a merge PR to handle these changes outside of this PR. This PR can be tested once that's through.

**Comment by demory** [comment]
Waiting for this PR to be merged: https://github.com/ibi-group/datatools-ui/pull/909

**Comment by philip-cline** [comment]
This should be good 2 go for review @br648

File: lib/editor/components/timetable/TimetableEditor.js [hunk 1]
```diff
@@ -219,8 +219,12 @@ export default class TimetableEditor extends Component<Props, State> {
         // }
         // if (stop.timepoint === null || stop.timepoint) {
         objectPath.set(newRow, `stopTimes.${i}.departureTime`, patternHaltIsStop(halt) ? cumulativeTravelTime : null)
+        objectPath.set(newRow, `stopTimes.${i}.pickupType`, halt.pickupType)
+        objectPath.set(newRow, `stopTimes.${i}.dropOffType`, halt.dropOffType)
         objectPath.set(newRow, `stopTimes.${i}.continuousPickup`, halt.continuousPickup)
         objectPath.set(newRow, `stopTimes.${i}.continuousDropOff`, halt.continuousDropOff)
+        objectPath.set(newRow, `stopTimes.${i}.pickupBookingRuleId`, halt.pickupBookingRuleId)
+        objectPath.set(newRow, `stopTimes.${i}.dropOffBookingRuleId`, halt.dropOffBookingRuleId)
```
Comment (br648) [inline lib/editor/components/timetable/TimetableEditor.js:237]: I'm now seeing the same issue with "stop_headsign". Assuming it needs added here as well?

**Review by br648 — CHANGES_REQUESTED** [review]
Stop headsign needs adding now.

**Comment by binh-dam-ibigroup** [comment]
This should work now with the latest `dev-flex` code, @br648.

File: lib/editor/components/timetable/TimetableEditor.js (hunk 1 above)
Comment (br648) [reply lib/editor/components/timetable/TimetableEditor.js:237]: @philip-cline Now that dev-flex has been merged into this branch, should the stop_headsign parameter be present here now? I'm not seeing it so concerned this is still going to fail.

Comment (binh-dam-ibigroup) [reply lib/editor/components/timetable/TimetableEditor.js:237]: @br648 Are you referring to changes from #910?

Comment (br648) [reply lib/editor/components/timetable/TimetableEditor.js:237]: @binh-dam-ibigroup Afraid not. https://github.com/ibi-group/datatools-ui/pull/910 doesn't contain an update to the TimetableEditor. I would expect to see something like the following:

`objectPath.set(newRow, `stopTimes.${i}.stop_headsign`, halt.stop_headsign)`

Comment (binh-dam-ibigroup) [reply lib/editor/components/timetable/TimetableEditor.js:237]: @br648 Then it looks like you are referring to #890, which looking at the commit history, is not in the `dev-flex` branch.

Comment (miles-grant-ibigroup) [reply lib/editor/components/timetable/TimetableEditor.js:237]: @philip-cline and I will merge `dev` into `dev-flex` soon -- there are some merge conflicts we need to block out time to resolve

**Review by br648 — APPROVED** [review]
This is fine and can be merged with the caveat that the issue with stop headsign will be addressed in another PR.

**Review by philip-cline — APPROVED** [review]

**Pull request merged**